File: Semester 5/Mobile Application Programming/Server/citybreak/src/main/java/com/example/citybreak/model/CityBreak.java
package com.example.citybreak.model;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "citybreak")

public class CityBreak {

    @Id
    private Long id;

    @Column(name="city")
    private String city;

    @Column(name="country")
    private String country;

    @Column(name="startDate")
    private String startDate;

    @Column(name="endDate")
    private String endDate;

    @Column(name="description")
    private String description;

    @Column(name="accommodation")
    private String accommodation;

    @Column(name="budget")
    private String budget;

}

File: Semester 5/Mobile Application Programming/CityBreakAppNative/app/src/main/java/com/example/citybreakappnative/model/CityBreak.kt
package com.example.citybreakappnative.model
import android.os.Parcel
import android.util.Log
import android.os.Parcelable
import android.text.Editable
import kotlinx.android.parcel.Parcelize
//import org.threeten.bp.LocalDate
//import kotlinx.parcelize.Parcelize
import java.time.LocalDate

@Parcelize
data class CityBreak(
    var city:String, var country: String, var startDate :LocalDate,
    var endDate: LocalDate, var description: String, var accommodation: String, var budget: Int,var id: Int): Parcelable{
    companion object{
        var currentId = 0
    }
    constructor(city: String, country:String, startDate:LocalDate, endDate:LocalDate, description: String, accommodation: String, budget: Int): this(city,country, startDate, endDate, description, accommodation, budget,
        currentId++){
        Log.i("Model CityBreak Class: ", "CurrentId is $currentId")
    }

    override fun describeContents(): Int {
        TODO("Not yet implemented")
    }


}


